### PR TITLE
feat: harden against invalid license identifiers

### PR DESCRIPTION
**Pull request by mvayngrib**
-   add breaking test for invalid license string
-   harden `spdx-satisifes` usage

example parse failure
```
2024-04-25T18:49:03.253Z license-compliance:license failed to parse license identifier "MIT or Apache-2.0" Error: Unexpected `o` at offset 4
    at module.exports (/Users/mv/code/license-compliance/node_modules/spdx-satisfies/node_modules/spdx-expression-parse/scan.js:124:13)
    at module.exports (/Users/mv/code/license-compliance/node_modules/spdx-satisfies/node_modules/spdx-expression-parse/index.js:7:16)
    at satisfies (/Users/mv/code/license-compliance/node_modules/spdx-satisfies/index.js:134:45)
```

### CODE REVIEW

File: src/license.ts [hunk 1]
```diff
@@ -6,12 +6,21 @@ import chalk from "chalk";
 import Debug from "debug";
 import path from "path";
 import parse from "spdx-expression-parse";
-import satisfies from "spdx-satisfies";
+import spdxSatisfiesOriginal from "spdx-satisfies";
```
Comment (tmorell) [inline src/license.ts:9]: @mvayngrib Rename to `spdxSatisfies`

File: src/license.ts [hunk 2]
```diff
@@ -6,12 +6,21 @@ import chalk from "chalk";
...
 
 import { LicenseStatus, Literals } from "./enumerations";
 import { Configuration, License, NpmPackage, OldLicenseFormat, Package } from "./interfaces";
 import * as util from "./util";
 
+const satisfies = (spdxLicense: string, str: string) => {
```
Comment (tmorell) [inline src/license.ts:15]: @mvayngrib Rename to `(allowedLicenses: string, license: string)`

**Review by tmorell — CHANGES_REQUESTED** [review]
@mvayngrib Thank you for the PR. It is an improvement to wrap the `satisfies` function.

File: src/license.ts (hunk 1 above)
Comment (mvayngrib) [reply src/license.ts]: 5d359a8442bd0f286ff7ee853962a42231653be5

File: src/license.ts (hunk 2 above)
Comment (mvayngrib) [reply src/license.ts]: 720b47559dad389bdacbd9490e03cfb4ae3faa1f

**Review by tmorell — APPROVED** [review]

**Pull request merged**